lib/concurrent/atomic/atomic_fixnum.rb

```diff
@@ -74,10 +68,7 @@ def value=(value)
     #
     #   @return [Fixnum] the current value after incrementation
     def increment
-      @mutex.lock
-      @value += 1
-    ensure
-      @mutex.unlock
+      synchronize { @value += 1 }
```

Comment:
I did some performance testing of `FixNum` and `BigDecimal` once and `BigDecimal` is *orders of magnitude* slower. I think it's OK for us to place a cap on this class and let the runtime throw an exception if it gets too big.